### DISCUSSION TITLE
Error while compiling LDT

### DISCUSSION THREAD

**Discussion by MariRiba** [NASA-LIS/LISF#1380] [Q&A]
Hi , I am trying to install LISF (downloaded from "git clone https://github.com/NASA-LIS/LISF") master branch.
I have installed all the dependencies and successfully configured LDT.
But I am getting this error while compiling it.

Compiling LDT

[INFO] Compiling LDT source code

gcc -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/ -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/ -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/ -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/ -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_mpiMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/ -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/ -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/ -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/ -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_logMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/ -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/ -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/ -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/ -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_timeMgrMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/ -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/ -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/ -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/ -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_coreMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/ -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/ -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/ -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/ -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../core/LDT_DAobsDataMod.F90
gcc -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/ -I/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build//include/ -I/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2//include/ -I/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0//include/ -I/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4//include/ -I/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I/home/mari/Desktop/project/Dependencies/fortrangis-2.5//include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../obsSim/OSSEmask/AMSR2/AMSR2OSSEmask_Mod.F90
../obsSim/OSSEmask/AMSR2/AMSR2OSSEmask_Mod.F90:24:7:

24 | use LDT_constantsMod, only : LDT_CONST_PATH_LEN
| 1
Fatal Error: Cannot open module file ‘ldt_constantsmod.mod’ for reading at (1): No such file or directory
compilation terminated.
make: *** [Makefile:108: AMSR2OSSEmask_Mod.o] Error 1
[ERR] Compile failed

My configure
(vvv) root@mari-VivoBook-ASUSLaptop-X509DA-M509DA:/home/mari/Desktop/project/LISF/ldt# ./configure
Setting up configuration for LDT
Parallelism (0-serial, 1-dmpar, default=0):
Optimization level (-3=strict checks with warnings, -2=strict checks, -1=debug, 0,1,2,3, default=2):
Assume little/big_endian data format (1-little, 2-big, default=2):
Use GRIBAPI/ECCODES? (0-neither, 1-gribapi, 2-eccodes, default=2):
NETCDF version (3 or 4, default=4)?:
NETCDF use shuffle filter? (1-yes, 0-no, default = 1):
NETCDF use deflate filter? (1-yes, 0-no, default = 1):
NETCDF use deflate level? (1 to 9-yes, 0-no, default = 9):
Use HDF4? (1-yes, 0-no, default=1):
Use HDF5? (1-yes, 0-no, default=1):
Use HDFEOS? (1-yes, 0-no, default=1):
Enable GeoTIFF support? (1-yes, 0-no, default=1):
Enable LIBGEOTIFF support? (1-yes, 0-no, default=1):
Include date/time stamp history? (1-yes, 0-no, default=1):
configure.ldt file generated successfully

Settings are written to configure.ldt in the make directory.
If you wish to change settings, please edit that file.
To compile, run the compile script.

My env variables path are as follows:

gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0

export LDT_ARCH=linux_gfortran
export LDT_FC=gcc
export LDT_CC=gcc
export LDT_MODESMF=/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/
export LDT_LIBESMF=/home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/
export LDT_OPENJPEG=/home/mari/Desktop/project/Dependencies/openjpeg-2.4.0/src/
export LDT_ECCODES=/home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build/
export LDT_NETCDF=/home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2/
export LDT_HDF4=/home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4/
export LDT_HDF5=/home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5/
export LDT_GDAL=/home/mari/Desktop/project/Dependencies/
export LDT_HDFEOS=/home/mari/Desktop/project/Dependencies/hdf-eos2-3.0/
export LDT_FORTRANGIS=/home/mari/Desktop/project/Dependencies/fortrangis-2.5/
export LDT_LIBGEOTIFF=/home/mari/Desktop/project/Dependencies/libgeotiff-master/libgeotiff/
export LDT_JPEG=/usr/lib/x86_64-linux-gnu/

And my configure.ldt
FC = gcc
FC77 = gcc
LD = gcc
CC = gcc
AR = ar
MOD_ESMF = /home/mari/Desktop/project/Dependencies/esmf/mod/modO/Linux.gfortran.64.mpiuni.default/
LIB_ESMF = /home/mari/Desktop/project/Dependencies/esmf/lib/libO/Linux.gfortran.64.mpiuni.default/
INC_JPEG2000 = /home/mari/Desktop/project/Dependencies/openjpeg-2.4.0/include/
LIB_JPEG2000 = /home/mari/Desktop/project/Dependencies/openjpeg-2.4.0/lib/
INC_ECCODES = /home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build/include/
LIB_ECCODES = /home/mari/Desktop/project/Dependencies/eccodes-2.30.2-Source/build/lib/
INC_NETCDF = /home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2/include/
LIB_NETCDF = /home/mari/Desktop/project/Dependencies/netcdf-c-4.9.2/lib/
INC_HDF4 = /home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4/include/
LIB_HDF4 = /home/mari/Desktop/project/Dependencies/hdf-4.2.16/hdf4/lib/
INC_HDF5 = /home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5/include/
LIB_HDF5 = /home/mari/Desktop/project/Dependencies/hdf5-1.14.1-2/built-fortran/hdf5/lib/
INC_HDFEOS = /home/mari/Desktop/project/Dependencies/hdf-eos5-2.0/include/
LIB_HDFEOS = /home/mari/Desktop/project/Dependencies/hdf-eos5-2.0/lib/
INC_FORTRANGIS1 = /home/mari/Desktop/project/Dependencies/fortrangis-2.5/include/
INC_FORTRANGIS2 = /home/mari/Desktop/project/Dependencies/fortrangis-2.5/include/
LIB_FORTRANGIS = /home/mari/Desktop/project/Dependencies/fortrangis-2.5/lib/
INC_LIBGEOTIFF = /home/mari/Desktop/project/Dependencies/libgeotiff-master/libgeotiff/include/
LIB_LIBGEOTIFF = /home/mari/Desktop/project/Dependencies/libgeotiff-master/libgeotiff/lib/
LIB_GDAL = /home/mari/Desktop/project/Dependencies/gdal-3.7.0/lib/
CFLAGS = -c -DGFORTRAN -I$(INC_LIBGEOTIFF)
FFLAGS77 = -c -pass-exit-codes -O2 -fconvert=big-endian -DGFORTRAN -I$(MOD_ESMF) -I$(INC_ECCODES) -I$(INC_NETCDF) -I$(INC_HDFEOS) -I$(INC_HDF4) -I$(INC_HDF5) -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
FFLAGS = -c -pass-exit-codes -ffree-line-length-0 -O2 -fconvert=big-endian -DGFORTRAN -I$(MOD_ESMF) -I$(INC_ECCODES) -I$(INC_NETCDF) -I$(INC_HDFEOS) -I$(INC_HDF4) -I$(INC_HDF5) -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
LDFLAGS = -L$(LIB_ESMF) -lesmf -lstdc++ -lz -L$(LIB_ECCODES) -leccodes_f90 -leccodes -L$(LIB_JPEG2000) -lopenjp2 -L$(LIB_NETCDF) -lnetcdff -lnetcdf -L$(LIB_HDFEOS) -lhdfeos -lGctp -L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz -L$(LIB_HDF5) -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl -L$(LIB_FORTRANGIS) -lfortrangis -lfortranc -L$(LIB_GDAL) -lgdal -L$(LIB_LIBGEOTIFF) -ltiff -lgeotiff -lm -lz -ljpeg

Please Help !!!!!!!!

**Comment by sudarsan3010** [comment]
Hi,

In my case I have GNU compiler and for linux_gfortran architecture, I gave 'gfortran' and 'gcc' as below,

export LDT_ARCH=linux_gfortran
export LDT_FC=gfortran
export LDT_CC=gcc

(GNU compiler)
or if you are compiling with mpich then,
export LDT_FC=mpif90
export LDT_CC=mpicc

and in bashrc:

export CC=gcc
export CXX=g++
export FC=gfortran
export FCFLAGS=-m64
export F77=gfortran
export FFLAGS=-m64
export C90=c90

**Reply by MariRiba** [reply]
Thanks for the tips
I tried reinstalling all the dependencies and again tried to run the configure.
This time i tried both
export LDT_ARCH=linux_gfortran
export LDT_FC=gfortran
export LDT_CC=gcc

and 
export LDT_FC=mpif90
export LDT_CC=mpicc

and i can see that I used export LDT_FC=gcc in my previous configuration which might be wrong.
Anyway I tried again and this time I got different error.

gfortran -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Documents/dep/esmf-8.5.0/mod/modO/Linux.gfortran.64.mpiuni.default  -I/home/mari/Documents/dep/eccodes-2.31.0-Source/build/include/ -I/home/mari/Documents/dep/netcdf-fortran-4.6.1/build/include/  -I/home/mari/Documents/dep/hdf-eos2-3.0/build/include/  -I/home/mari/Documents/dep/hdf-4.2.16-2/build/include/  -I/home/mari/Documents/dep/hdf5-1.14.1-2/build-fortran/include/  -I/home/mari/Documents/dep/fortrangis/build/include/ -I/home/mari/Documents/dep/fortrangis/build/include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../DAobs/GLASSlai/readGLASSlaiObs.F90
../DAobs/GLASSlai/readGLASSlaiObs.F90:194:55:

  168 |      iret = gdrdfld(grid_id,lai_name,start,stride,edge,lai_raw_avhrr)
      |                                                       2
......
  194 |      iret = gdrdfld(grid_id,lai_name,start,stride,edge,lai_raw_modis)
      |                                                       1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(1)/INTEGER(2)).
make: *** [Makefile:108: readGLASSlaiObs.o] Error 1

[ERR] Compile failed
-----------------------------------------------------------------------------------------------------------------------------
And I am stuck here .

**Reply by sudarsan3010** [reply]
Which version of LISF are you compiling?

It successfully compiled for me by using (I used the latest LISF 7.4.2)
export LDT_FC=mpif90
export LDT_CC=mpicc

and also in the same way the ESMF should also be compiled by,
export ESMF_COMM=mpich
export ESMF_BOPT=g
(add this env variables before gmake); if you want to use mpich in LISF

For your current error you can
cd make (from ldt folder)
vi Makefile (in this file you can find that line readGLASSlaiObs.o)
try to remove " -fallow-argument-mismatch " from that line and the next lines also.

It worked for me.

**Reply by MariRiba** [reply]
Hi 
I am also using the latest LISF 7.4.2
and I used 
export LDT_FC=mpif90
export LDT_CC=mpicc

and also I compiled the ESMF using 
export ESMF_COMM=mpich
export ESMF_BOPT=g

before using gmake

and then I removed the "-fallow-argument-mismatch " from Makefile , which now looks like

`ifeq ($(LIS_ARCH),linux_gfortran)
# For MPI
LDT_historyMod.o: LDT_historyMod.F90
        $(FC) $(FFLAGS) $(HEADER_DIRS) $<
# For HDF-EOS2
readGLASSlaiObs.o: readGLASSlaiObs.F90
        $(FC) $(FFLAGS) $(HEADER_DIRS) $<
readNASA_AMSREsmObs.o: readNASA_AMSREsmObs.F90
        $(FC) $(FFLAGS) $(HEADER_DIRS) $<
endif`

This looks different here , so i am attaching a screenshot photo of the above code.
![Screenshot from 2023-08-02 22-36-11](https://github.com/NASA-LIS/LISF/assets/138671730/39740355-3936-4317-8d8c-9398592fcbfc)

But still I am getting the same error.

The error is from "LISF_public_release_7.4.2/ldt/DAobs/GLASSlai/readGLASSlaiObs.F90" as shown in the compilation error.Can something be done from this file?
For reference I am uploading the error:

mpif90 -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I/home/mari/Documents/dep/esmf-8.5.0/DEFAULTINSTALLDIR/mod/modg/Linux.gfortran.64.mpich.default  -I/home/mari/Documents/dep/eccodes-2.31.0-Source/eccodes/include/ -I/home/mari/Documents/dep/netcdf-fortran-4.6.1/build/include/  -I/home/mari/Documents/dep/hdf-eos2-3.0/build/include/  -I/home/mari/Documents/dep/hdf-4.2.16-2/build/include/  -I/home/mari/Documents/dep/hdf5-1.14.1-2/build-fortran/include/  -I/home/mari/Documents/dep/fortrangis/build/include/ -I/home/mari/Documents/dep/fortrangis/build/include/ -I./ -I../lib/xmrg/ -I../params/VIC/ -I../USAFSI/ ../DAobs/GLASSlai/readGLASSlaiObs.F90
../DAobs/GLASSlai/readGLASSlaiObs.F90:194:55:

  168 |      iret = gdrdfld(grid_id,lai_name,start,stride,edge,lai_raw_avhrr)
      |                                                       2
......
  194 |      iret = gdrdfld(grid_id,lai_name,start,stride,edge,lai_raw_modis)
      |                                                       1
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(1)/INTEGER(2)).
make: *** [Makefile:110: readGLASSlaiObs.o] Error 1

[ERR] Compile failed

And also the configure file too:

FC              = mpif90
FC77            = mpif90
LD              = mpif90
CC              = mpicc
AR              = ar
MOD_ESMF        = /home/mari/Documents/dep/esmf-8.5.0/DEFAULTINSTALLDIR/mod/modg/Linux.gfortran.64.mpich.default
LIB_ESMF        = /home/mari/Documents/dep/esmf-8.5.0/DEFAULTINSTALLDIR/lib/libg/Linux.gfortran.64.mpich.default
INC_JPEG2000      = /home/mari/Documents/dep/openjpeg-2.5.0/include/
LIB_JPEG2000      = /home/mari/Documents/dep/openjpeg-2.5.0/lib/
INC_ECCODES     = /home/mari/Documents/dep/eccodes-2.31.0-Source/eccodes/include/
LIB_ECCODES     = /home/mari/Documents/dep/eccodes-2.31.0-Source/eccodes/lib/
INC_NETCDF      = /home/mari/Documents/dep/netcdf-fortran-4.6.1/build/include/
LIB_NETCDF      = /home/mari/Documents/dep/netcdf-fortran-4.6.1/build/lib/
INC_HDF4        = /home/mari/Documents/dep/hdf-4.2.16-2/build/include/
LIB_HDF4        = /home/mari/Documents/dep/hdf-4.2.16-2/build/lib/
INC_HDF5        = /home/mari/Documents/dep/hdf5-1.14.1-2/build-fortran/include/
LIB_HDF5        = /home/mari/Documents/dep/hdf5-1.14.1-2/build-fortran/lib/
INC_HDFEOS      = /home/mari/Documents/dep/hdf-eos2-3.0/build/include/
LIB_HDFEOS      = /home/mari/Documents/dep/hdf-eos2-3.0/build/lib/
INC_FORTRANGIS1 = /home/mari/Documents/dep/fortrangis/build/include/
INC_FORTRANGIS2 = /home/mari/Documents/dep/fortrangis/build/include/
LIB_FORTRANGIS  = /home/mari/Documents/dep/fortrangis/build/lib/
INC_LIBGEOTIFF  = /home/mari/Documents/dep/libgeotiff/libgeotiff/build/include/
LIB_LIBGEOTIFF  = /home/mari/Documents/dep/libgeotiff/libgeotiff/build/lib/
LIB_GDAL        = /home/mari/Documents/dep/gdal/build/lib/
CFLAGS          = -c -DGFORTRAN  -I$(INC_LIBGEOTIFF)
FFLAGS77        = -c -pass-exit-codes -O2   -fconvert=big-endian -DGFORTRAN -I$(MOD_ESMF)  -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)  -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
FFLAGS          =  -c -pass-exit-codes -ffree-line-length-0 -O2   -fconvert=big-endian -DGFORTRAN -I$(MOD_ESMF)  -I$(INC_ECCODES) -I$(INC_NETCDF)  -I$(INC_HDFEOS)  -I$(INC_HDF4)  -I$(INC_HDF5)  -I$(INC_FORTRANGIS1) -I$(INC_FORTRANGIS2)
LDFLAGS         =  -L$(LIB_ESMF) -lesmf -lstdc++ -lz  -L$(LIB_ECCODES) -leccodes_f90 -leccodes -L$(LIB_JPEG2000) -lopenjp2 -L$(LIB_NETCDF) -lnetcdff -lnetcdf -L$(LIB_HDFEOS) -lhdfeos -lGctp -L$(LIB_HDF4) -lmfhdf -ldf -ljpeg -lz  -L$(LIB_HDF5) -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl -L$(LIB_FORTRANGIS) -lfortrangis -lfortranc -L$(LIB_GDAL) -lgdal -L$(LIB_LIBGEOTIFF)  -ltiff -lgeotiff -lm -lz -ljpeg 

I am using root to install all these , can it be the problem?

**Comment by sudarsan3010** [comment]
In configuration file
LIB_ECCODES     = /home/mari/Documents/dep/eccodes-2.31.0-Source/eccodes/lib/
confirm whether the library files are present at this location, usually eccodes library get installed at lib64 (for 64-bit os)

And while configuring LDT remove HDF4 and HDFEOS.. (According to makefile HDFEOS should be causing problem). First compile only with the necessary libraries, if it is done then you can recompile with all these optional libraries.

**Comment by jvgeiger** [comment]
Hello,

Thanks @sudarsan3010 for providing some help.  Disabling HDF4 and HDFEOS2 at the `./configure` step is a good suggestion.


@MariRiba, to compile with the GNU compilers, you do need the `-fallow-argument-mismatch` option.  Now that you have fixed your environment settings, let's start over again. Please

```
cd ldt/make
git restore Makefile
make realclean
```

Then

```
cd ..
./configure
./compile
```

Whenever you change environment variables or rerun `./configure`, you should run "make realclean" to be safe.

**Reply by MariRiba** [reply]
Hi @jvgeiger 
I was trying to install LISF in my laptop and I think there were many problems related to it that caused difficulty in its installation. So I am switching to a HPC and instead of just LISF , I think I will install NU-WRF, which also contains LISF.
Thank you for your help.
I think I will have to close this discussion .
